### ISSUE TITLE
Crash file generated and no data in the nibetaseries folder

### ISSUE THREAD

**Issue by LilianYou** [HBClab/NiBetaSeries#319]
Hi,

I have tried to replicate the folder format from the tutorial dataset. However, each time I run the beta series code, I only got a 'logs' folder under the 'nibetaseries' folder with no subject folder. In addition, I got subject folder under derivatives, the folder only contains one 'log' folder that has a crash file inside. 

Here is my output:

```
The Example Command:
 nibs -c WhiteMatter CSF --participant-label 033 --estimator lsa --hrf-model glover -w /mnt/chrastil/lab/users/lily/localizer_beta_series_run_1/derivatives/work -a /mnt/chrastil/lab/users/lily/localizer_beta_series_run_1/derivatives/data/Schaefer2018_100Parcels_7Networks_order_FSLMNI152_2mm.nii.gz -l /mnt/chrastil/lab/users/lily/localizer_beta_series_run_1/derivatives/data/Schaefer2018_100Parcels_7Networks_order.tsv /mnt/chrastil/lab/users/lily/localizer_beta_series_run_1 fmriprep   /mnt/chrastil/lab/users/lily/localizer_beta_series_run_1/derivatives participant

b'200603-03:34:19,254 nipype.utils INFO:\n'
b'\t No new version available.\n'
b"/home/lily/.local/lib/python3.6/site-packages/bids/layout/models.py:157: UserWarning: Accessing entities as attributes is deprecated as of 0.7. Please use the .entities dictionary instead (i.e., .entities['task'] instead of .task.\n"
b'  % (attr, attr))\n'
b"/home/lily/.local/lib/python3.6/site-packages/bids/layout/models.py:157: UserWarning: Accessing entities as attributes is deprecated as of 0.7. Please use the .entities dictionary instead (i.e., .entities['run'] instead of .run.\n"
b'  % (attr, attr))\n'
b"/home/lily/.local/lib/python3.6/site-packages/bids/layout/models.py:157: UserWarning: Accessing entities as attributes is deprecated as of 0.7. Please use the .entities dictionary instead (i.e., .entities['space'] instead of .space.\n"
b'  % (attr, attr))\n'
b'200603-03:34:19,753 nipype.workflow INFO:\n'
b"\t Workflow nibetaseries_participant_wf settings: ['check', 'execution', 'logging', 'monitoring']\n"
b'200603-03:34:19,781 nipype.workflow INFO:\n'
b'\t Running in parallel.\n'
b'200603-03:34:19,785 nipype.workflow INFO:\n'
b'\t [MultiProc] Running 0 tasks, and 1 jobs ready. Free memory (GB): 226.69/226.69, Free processors: 32/32.\n'
b'200603-03:34:19,862 nipype.workflow INFO:\n'
b'\t [Node] Setting-up "nibetaseries_participant_wf.single_subject033_wf.betaseries_wf.betaseries_node" in "/mnt/chrastil/lab/users/lily/localizer_beta_series_run_1/derivatives/work/NiBetaSeries_work/nibetaseries_participant_wf/single_subject033_wf/betaseries_wf/d997438a133df12646f8b73af4436d4935dfdb15/betaseries_node".\n'
b'200603-03:34:19,872 nipype.workflow INFO:\n'
b'\t [Node] Running "betaseries_node" ("nibetaseries.interfaces.nistats.LSABetaSeries")\n'
b'200603-03:34:20,173 nipype.workflow WARNING:\n'
b'\t Storing result file without outputs\n'
b'200603-03:34:20,174 nipype.workflow WARNING:\n'
b'\t [Node] Error on "nibetaseries_participant_wf.single_subject033_wf.betaseries_wf.betaseries_node" (/mnt/chrastil/lab/users/lily/localizer_beta_series_run_1/derivatives/work/NiBetaSeries_work/nibetaseries_participant_wf/single_subject033_wf/betaseries_wf/d997438a133df12646f8b73af4436d4935dfdb15/betaseries_node)\n'
b'200603-03:34:21,787 nipype.workflow ERROR:\n'
b'\t Node betaseries_node.a0 failed to run on host Atlas.\n'
b'200603-03:34:21,788 nipype.workflow ERROR:\n'
b'\t Saving crash info to /mnt/chrastil/lab/users/lily/localizer_beta_series_run_1/derivatives/sub-033/log/crash-20200603-033421-lily-betaseries_node.a0-95ee80a4-69ff-4ad8-bb8f-63374449373e.txt\n'
b'Traceback (most recent call last):\n'
b'  File "/home/lily/.local/lib/python3.6/site-packages/nipype/pipeline/plugins/multiproc.py", line 148, in __init__\n'
b'    mp_context=mp_context,\n'
b"TypeError: __init__() got an unexpected keyword argument 'initializer'\n"
b'\n'
b'During handling of the above exception, another exception occurred:\n'
b'\n'
b'Traceback (most recent call last):\n'
b'  File "/home/lily/.local/lib/python3.6/site-packages/nipype/pipeline/plugins/multiproc.py", line 67, in run_node\n'
b'    result["result"] = node.run(updatehash=updatehash)\n'
b'  File "/home/lily/.local/lib/python3.6/site-packages/nipype/pipeline/engine/nodes.py", line 516, in run\n'
b'    result = self._run_interface(execute=True)\n'
b'  File "/home/lily/.local/lib/python3.6/site-packages/nipype/pipeline/engine/nodes.py", line 635, in _run_interface\n'
b'    return self._run_command(execute)\n'
b'  File "/home/lily/.local/lib/python3.6/site-packages/nipype/pipeline/engine/nodes.py", line 741, in _run_command\n'
b'    result = self._interface.run(cwd=outdir)\n'
b'  File "/home/lily/.local/lib/python3.6/site-packages/nipype/interfaces/base/core.py", line 397, in run\n'
b'    runtime = self._run_interface(runtime)\n'
b'  File "/home/lily/.local/lib/python3.6/site-packages/nibetaseries/interfaces/nistats.py", line 196, in _run_interface\n'
b"    t_r = self.inputs.bold_metadata['RepetitionTime']\n"
b"KeyError: 'RepetitionTime'\n"
b'\n'
b'200603-03:34:21,790 nipype.workflow INFO:\n'
b'\t [MultiProc] Running 0 tasks, and 0 jobs ready. Free memory (GB): 226.69/226.69, Free processors: 32/32.\n'
b'200603-03:34:23,787 nipype.workflow INFO:\n'
b'\t ***********************************\n'
b'200603-03:34:23,787 nipype.workflow ERROR:\n'
b'\t could not run node: nibetaseries_participant_wf.single_subject033_wf.betaseries_wf.betaseries_node.a0\n'
b'200603-03:34:23,787 nipype.workflow INFO:\n'
b'\t crashfile: /mnt/chrastil/lab/users/lily/localizer_beta_series_run_1/derivatives/sub-033/log/crash-20200603-033421-lily-betaseries_node.a0-95ee80a4-69ff-4ad8-bb8f-63374449373e.txt\n'
b'200603-03:34:23,787 nipype.workflow INFO:\n'
b'\t ***********************************\n'
b'Workflow did not execute cleanly\n'
```

I also attached my crash file here
[crash-20200603-033421-lily-betaseries_node.a0-95ee80a4-69ff-4ad8-bb8f-63374449373e.txt](https://github.com/HBClab/NiBetaSeries/files/4722743/crash-20200603-033421-lily-betaseries_node.a0-95ee80a4-69ff-4ad8-bb8f-63374449373e.txt)

By the way, I don't think there is anything wrong with my python packages since everything looks fine when I run the tutorial code on my Jupiter notebook.

Very appreciated,
Lily

**Comment by LilianYou** [comment]
By the way, I don't think there is anything wrong with my python packages since everything looks fine when I run the tutorial code on my Jupiter notebook.

**Comment by jdkent** [comment]
Hi Lily,

The generation of the crash files is a little convoluted right now, apologies.

```
"KeyError: 'RepetitionTime'\n"
```

This error message suggests the sidecar json file for your functional images in your BIDS dataset cannot find the key "RepetitionTime".
Here is the [relevant portion of the bids specification](https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/01-magnetic-resonance-imaging-data.html#required-fields)

An example json file could have these contents:
```json
{
    "RepetitionTime": 2,
    "TaskName": "boldRun1"
}
```

This file would be located either at the top level of your BIDS dataset, or adjacent to the nifti files.

This all assumes that you have access to the RAW BIDS dataset in addition to the derivatives dataset, however.

Let me know if this helps or if you need any additional clarification.

**Comment by LilianYou** [comment]
Dear expert,

I noticed the json file problem and moved the json file next to the nifti dataset. Here is a screenshot:

<img width="708" alt="Screen Shot 2020-06-10 at 3 49 48 AM" src="https://user-images.githubusercontent.com/30990195/84259958-17f2e280-aace-11ea-9034-3183f621d07f.png">

The json file has a lot of information, which includes the repetition time, task name, etc. Here is a link to the json file https://drive.google.com/file/d/1lu7qe5kEen4kuUSxcka5i5oFSW7eHFAm/view?usp=sharing . I'm wondering if the error is due to there being too much information in the json file? If yes, how should I clean it up?

I am looking forward to your reply!

Best,
Lily

**Comment by LilianYou** [comment]
There is no tree command in the terminal and I'm not the administrator so I can not install one. If you want to see a list of the file structure, here it is:

<img width="840" alt="Screen Shot 2020-06-10 at 11 52 07 PM" src="https://user-images.githubusercontent.com/30990195/84354320-83dc5600-ab75-11ea-988e-1dc0402bc7f8.png">
<img width="601" alt="Screen Shot 2020-06-10 at 11 52 15 PM" src="https://user-images.githubusercontent.com/30990195/84354329-863eb000-ab75-11ea-9486-c1e19c19e8bf.png">

**Comment by LilianYou** [comment]
I moved the json file under the func folder, and the process gets a bit further. However, there is still a crash, with error named 'Region'. Here are the crash files: https://drive.google.com/drive/folders/1wo-Jk7JYBPedz577hiHlbCDcyNEFemFP?usp=sharing

Is it still a naming issue?

**Comment by jdkent** [comment]
Awesome, looks like those file naming issues are resolved.

The error you're seeing now deals with the expected headings for the [atlas lookup table](https://nibetaseries.readthedocs.io/en/stable/auto_examples/plot_run_nibetaseries.html#rename-columns-with-the-approved-headings-index-and-regions).

If you are using `Schaefer2018_100Parcels_7Networks_order.tsv` as your lookup table in your `nibs` command, the file **must** have `index` and `regions` as column names.

**Comment by LilianYou** [comment]
Hi James,

Yes, the problem is solved and I can run the code successfully! Do you mind if I ask you an additional question? Here is my correlation plot of four conditions. It seems the correlation just increases from the top condition to the lower condition. I wasn't expecting this because these four conditions are just four categories at the same semantic level. I'm new to beta series analysis. Does it look normal to you? 

<img width="266" alt="Screen Shot 2020-06-11 at 3 55 08 PM" src="https://user-images.githubusercontent.com/30990195/84447253-5edef600-abfc-11ea-9a8e-b66091bc77b4.png">

**Comment by jdkent** [comment]
Yay!

I can only offer my own limited experience, but having this many correlations close to 1 suggests to me that there may be few events and they may be close together. How many events do you have for the N, S, W, and E conditions and on average how much time separates one stimulus to the next?

**Comment by LilianYou** [comment]
There are around 15-20 events for each condition. The separate time is around 1s.

**Comment by jdkent** [comment]
Thanks for the info, and I just noticed that `RepetitionTime` was defined twice in the `json` file you shared, one saying 0.72 and the other saying 720, BIDS wants the `RepetitionTime` stated in seconds, so I presume the 0.72 is the correct one (you may want to rerun after remove the duplicate key).

But with only a second between stimuli, even with a repetition time of 0.72 seconds, the BOLD responses would be severely overlapping making it difficult to reliably measure separate task conditions.

With that in mind, I would expect to see extreme correlations with this dataset, so I would probably expect to see what you showed me. (but do re-run after removing the duplicate key)

**Comment by LilianYou** [comment]
Hi James,

Nice catch. I removed the duplicate 'repetition time' key and reran the code. I got the same answer though. I think your explanation about BOLD signal overlapping makes sense to me. 

Thanks,
Lily
